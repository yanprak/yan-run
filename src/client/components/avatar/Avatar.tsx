import React from 'react';
import {Props} from './types';
import './avatar.scss';

const Avatar: Props = ({url, size, ...otherProps}) => {
  const avatarStyle = url ? { backgroundImage: `url(${url})` } : {};
  return (
    <div className={`avatar avatar_${size ? size : 'medium'}`} style={avatarStyle} {...otherProps}></div>
  );
};

export default Avatar;
