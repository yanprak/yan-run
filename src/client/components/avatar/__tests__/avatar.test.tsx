import React from 'react';
import TestRenderer from 'react-test-renderer';
import Avatar from '../Avatar';

test('Avatar test', () => {
  const component = TestRenderer.create(
    <Avatar url="/avatar.png" size="small" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
