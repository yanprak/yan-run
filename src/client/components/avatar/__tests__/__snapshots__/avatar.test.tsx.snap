// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar test 1`] = `
<div
  className="avatar avatar_small "
  style={
    Object {
      "backgroundImage": "url(/avatar.png)",
    }
  }
/>
`;
